### PR TITLE
Switch to low level Elasticsearch Library

### PR DESCRIPTION
**Pull request by ghickman**
This rebuilds the existing Elasticsearch work using the elasticsearch library, dropping all uses of the elasticsearch-dsl library.

#### Why?
Elasticsearch-DSL, while an official Elastic library, tended to hide the underlying concepts from the reader while also mixing some of those concepts together (a DocType subclass both initialises an index and searches it).

The elasticsearch library is the low level wrapper Elastic provides, but it's API isn't necessarily the easiest to work with.  However it has the benefit of allowing the user to pass dictionaries that match the official Elasticsearch docs.  I've created a thin wrapper around this library to try and improve the usability of it (particularly around bulk importing) in a similar theme to [pyelasticsearch](https://pypi.org/project/pyelasticsearch/) which is currently locked to `elasticsearch>=1.3.0,<2.0.0`.

#### Services Pattern
I've wrapped up this new block of work in a services module. The idea is to have each service provide a simple way to access their connection and any helpers related to them in a [hopefully!] intuitive import structure.

### CODE REVIEW

File: treeherder/model/models.py [hunk 1]
```diff
@@ -18,9 +18,11 @@
 from django.forms import model_to_dict
 from django.utils import timezone
 from django.utils.encoding import python_2_unicode_compatible
+from more_itertools import flatten
```
Comment (edmorley) [inline treeherder/model/models.py]: The `flatten` method is pretty simple:
https://github.com/erikrose/more-itertools/blob/96c4136dd4cb47a18f2522dff6bb113c9bf36860/more_itertools/recipes.py#L236-L245
(and actually comes from https://docs.python.org/2/library/itertools.html#recipes)

I'm not 100% against adding a dependency on [more_itertools](https://github.com/erikrose/more-itertools), however curious to hear your thoughts for and against?

Comment (ghickman) [reply treeherder/model/models.py]: I always forget how simple `flatten` is!

re the recipe: more-itertools packages all the recipes from there with some extras thrown in, it's really useful!

re adding it as a dep: given how simple flatten is, I'm not fussed about including the whole package.  On the one hand it's very stable and I regularly find myself using a few methods from it, on the other it's another dep with upgrade overhead.

Comment (edmorley) [reply treeherder/model/models.py]: Perhaps we can save adding the dep until a time where we use a second method from the package?

Comment (ghickman) [reply treeherder/model/models.py]: Updated and rebased!

File: treeherder/services/elasticsearch/connection.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+from django.conf import settings
+from elasticsearch import Elasticsearch
+
+url = settings.ELASTICSEARCH_URL
+if url:
+    es_conn = Elasticsearch(settings.ELASTICSEARCH_URL)
```
Comment (edmorley) [inline treeherder/services/elasticsearch/connection.py]: `s/settings.ELASTICSEARCH_URL/url/` (sorry for not spotting before)

Or possibly one step further: `es_conn = Elasticsearch(url) if url else None`
(up to you; guess with Lazy might change soon anyway)

**Pull request merged**